scipy/ndimage/filters.py

```diff
@@ -1367,6 +1367,16 @@ def median_filter(input, size=None, footprint=None, output=None,
     median_filter : ndarray
         Filtered array. Has the same shape as `input`.
 
+    See also
+    --------
+    scipy.signal.medfilt2d
+
+    Notes
+    -----
+    For 2-dimensional images with `np.ubyte`, `np.single` or `np.double`
+    datatypes the specialised function `scipy.signal.medfilt2d` is faster but
```

Comment:
I would say "may be faster" instead of "is faster" here. I got similar 10% speedup for float32 with `medfilter2d`, but actually measured `median_filter` to be ~1% faster for float64. Given these relatively small differences of within ~10% I don't think we can make definitive statements across hardware platforms.

some results from your example in #13509 run using the `%timeit` magic in an ipython console.

```Python
sig = sig.astype(np.float32)

%timeit medfilt2d(sig, (9, 9))
# 102 ms ± 1.05 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit median_filter(sig, (9, 9), mode='constant')
# 113 ms ± 1.47 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

sig = sig.astype(np.float64)

%timeit medfilt2d(sig, (9, 9))
# 111 ms ± 549 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit median_filter(sig, (9, 9), mode='constant')
# 110 ms ± 339 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```